circuit_knitting/cutting/cut_finding/cutting_actions.py

```diff
@@ -0,0 +1,436 @@
+
+        return [new_state]
+
+    @staticmethod
+    def get_cost_params(
+        gate_spec: GateSpec,
+    ) -> tuple[float | None, int, float | None]:
+        """
+        Get the cost parameters for gate cuts.
+
+        This method returns a tuple of the form:
+            (<gamma_lower_bound>, <num_bell_pairs>, <gamma_upper_bound>)
+
+        Since CKT does not support LOCC at the moment, these tuples will be of
+        the form (gamma, 0, gamma).
+        """
+        gate = gate_spec.gate
+        gamma = gate.gamma
+        return (gamma, 0, gamma)
+
+    def export_cuts(
+        self,
+        circuit_interface: SimpleGateList,
+        wire_map: list[Hashable],
+        gate_spec: GateSpec,
+        args,
+    ) -> None:
+        """Insert an LO gate cut into the input circuit for the specified gate and cut arguments."""
+        # pylint: disable=unused-argument
+        circuit_interface.insert_gate_cut(gate_spec.instruction_id, "LO")
+
+
+# Add ActionCutTwoQubitGate to the global variable disjoint_subcircuit_actions
+disjoint_subcircuit_actions.define_action(ActionCutTwoQubitGate())
+
+
+class ActionCutLeftWire(DisjointSearchAction):
+    """Cut the left (first input) wire of a two-qubit gate."""
+
+    def get_name(self) -> str:
+        """Return the look-up name of :class:`ActionCutLeftWire`."""
+        return "CutLeftWire"
+
+    def get_group_names(self) -> list[str]:
+        """Return the group name of :class:`ActionCutLeftWire`."""
+        return ["WireCut", "TwoQubitGates"]
+
+    def next_state_primitive(
+        self,
+        state: DisjointSubcircuitsState,
+        gate_spec: GateSpec,
+        max_width: int,
+    ) -> list[DisjointSubcircuitsState]:
+        """Return the state that results from cutting the left (first input) wire of the gate given by ``gate_spec``."""
+        gate = gate_spec.gate
+
+        # Cutting of multi-qubit gates is not supported in this release.
+        if len(gate.qubits) != 2:  # pragma: no cover
+            raise ValueError(
+                "In the current version, only the cutting of two qubit gates is supported."
+            )
+
+        # If the wire-cut limit would be exceeded, return the empty list.
+        if not state.can_add_wires(1):
+            return list()
+
+        q1 = gate.qubits[0]
+        q2 = gate.qubits[1]
+        w1 = state.get_wire(q1)
+        r1 = state.find_qubit_root(q1)
+        r2 = state.find_qubit_root(q2)
+
+        if r1 == r2:
+            return list()
+
+        if not state.can_expand_subcircuit(r2, 1, max_width):
+            return list()
+
+        new_state = state.copy()
+
+        rnew = new_state.new_wire(q1)
+        new_state.merge_roots(rnew, r2)
+        new_state.assert_donot_merge_roots(r1, r2)  # Because r2 < rnew
+
+        new_state.bell_pairs = cast(list, new_state.bell_pairs)
+        new_state.bell_pairs.append((r1, r2))
+        new_state.gamma_UB = cast(float, new_state.gamma_UB)
+        new_state.gamma_UB *= 4
```

Comment:
Just glancing at this, it's not clear to me why there's a multiplication by 4 here.